spec/mongoid/document_spec.rb

```diff
@@ -513,6 +513,22 @@ class << self; attr_accessor :name; end
         expect(church.as_json.keys.sort).to eq(%w(_id location name))
       end
 
+      context 'deprecation' do
+        let(:church) do
+          Church.create!(name: 'St. Basil')
+        end
+
+        it 'logs a deprecation warning when :compact is given' do
+          expect_any_instance_of(Logger).to receive(:warn).with('#as_json :compact option is deprecated. Please call #compact on the returned Hash object instead.')
+          church.as_json(compact: true)
+        end
+
+        it 'does not log a deprecation warning when :compact is not given' do
+          expect_any_instance_of(Logger).to_not receive(:warn)
```

Comment:
MINOR: I think this may be a potentially flaky test. Any warning logged anywhere in Mongoid will make this test fail. I believe it will be more robust (and, maybe, more readable) if we specify more explicitly what warning we should not receive.

Reply:
Fixed.